### ISSUE TITLE
Integration to Exim (implement Dovecot SASL endpoint)

### ISSUE THREAD

**Issue by digitive** [foxcpp/maddy#173]
Can Maddy be used as pure IMAP server, and work with Exim as smtp server?

**Comment by foxcpp** [comment]
It is possible to configure maddy to accept messages using LMTP protocol and deliver them to the local storage. 

Here is the configuration to do that:
```
lmtp unix:///run/maddy/lmtp.sock {
  deliver_to &local_mailboxes
}
```

It should be noted though, that IMAP part of maddy is much less mature than the SMTP part. Notably [this](https://github.com/emersion/go-imap/issues/318) issue means that multiple connections support is very broken and unsafe to use.

**Comment by foxcpp** [comment]
Additionally, you likely also want to have Exim check SMTP AUTH against the maddy users DB. Currently, there is no interface to do so, but there is no reason it cannot be added (I was thinking about adding a Dovecot SASL-compatible endpoint for this purpose)

**Comment by foxcpp** [comment]
According to https://wiki2.dovecot.org/HowTo/EximAndDovecotSASL, Exim supports Dovecot SASL protocol, so technically it is a de-facto standard for authentication delegation in the mail world. We can and probably should implement it.

**Comment by foxcpp** [comment]
Reference: https://wiki.dovecot.org/Design/AuthProtocol

**Comment by digitive** [comment]
Thanks @foxcpp. Supporting Dovecot SASL will be an awesome feature.

**Comment by foxcpp** [comment]
Client side implemented in e06f2e201c4db1f216affda99489f4415632efe5. Server side support coming soon :tm:.

**Comment by foxcpp** [comment]
Server side implementation available in the `dev` branch. Not considered stable until interoperability tests with https://github.com/albertito/chasquid (and possibly other MSAs) are a thing.
